cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java

```diff
@@ -0,0 +1,103 @@
+import org.opencds.cqf.fhir.cr.measure.common.MeasureEvalType;
+import org.opencds.cqf.fhir.utility.repository.InMemoryFhirRepository;
+
+class R4RepositorySubjectProviderTest {
+
+    private static final String PAT_ID_1 = "pat1";
+    private static final String PAT_ID_2 = "pat2";
+    private static final String ORG_ID_1 = "org1";
+    private static final String ORG_ID_2 = "org2";
+
+    private static final R4RepositorySubjectProvider TEST_SUBJECT_ENABLE_PART_OF =
+            new R4RepositorySubjectProvider(new SubjectProviderOptions().setPartOfEnabled(true));
+    private static final R4RepositorySubjectProvider TEST_SUBJECT_DISABLE_PART_OF =
+            new R4RepositorySubjectProvider(new SubjectProviderOptions().setPartOfEnabled(false));
+
+    private final Repository repository = new InMemoryFhirRepository(FhirContext.forR4Cached());
+
+    @BeforeEach
+    void beforeEach() {
+        final Organization org1 =
+                (Organization) new Organization().setId(new IdType(ResourceType.Organization.toString(), ORG_ID_1));
+        final IIdType orgId1 = repository.update(org1).getId().toUnqualifiedVersionless();
+
+        final Organization org2 = (Organization) new Organization()
+                .setPartOf(new Reference(orgId1.toUnqualifiedVersionless().getValue()))
+                .setId(new IdType(ResourceType.Organization.toString(), ORG_ID_2));
+
+        final IIdType orgId2 = repository.update(org2).getId().toUnqualifiedVersionless();
+
+        final Patient patient1 = new Patient();
+        patient1.setId(PAT_ID_1);
+        patient1.setManagingOrganization(
+                new Reference(orgId1.toUnqualifiedVersionless().getValue()));
+
+        final Patient patient2 = new Patient();
+        patient2.setId(PAT_ID_2);
+        patient2.setManagingOrganization(
+                new Reference(orgId2.toUnqualifiedVersionless().getValue()));
+
+        repository.update(patient1).getId().toUnqualifiedVersionless();
+        repository.update(patient2).getId().toUnqualifiedVersionless();
+    }
+
+    public static Stream<Arguments> getSubjectsParams() {
+        return Stream.of(
+                Arguments.of(
+                        TEST_SUBJECT_ENABLE_PART_OF,
+                        MeasureEvalType.SUBJECT,
+                        List.of(resourcify(ResourceType.Organization, ORG_ID_1)),
+                        // TODO:  LD:  this is technically incorrect:  it should be PAT_ID_1, PAT_ID_2
+                        // However, due to the fact that both InMemoryFhirRepository and IgRepository
+                        // do NOT support chained searches, the results can only be accurately verified
+                        // with a repository that supports DAOs.
+                        Stream.of(PAT_ID_1, PAT_ID_1)
+                                .map(id -> resourcify(ResourceType.Patient, id))
+                                .collect(Collectors.toList())),
```

Comment:
suggestion: Now that we are Java 17 compliant, we can replace this usage of 'Stream.collect(Collectors.toList())' with 'Stream.toList()' and ensure that the list is unmodified.